allennlp/models/semantic_role_labeler.py

```diff
@@ -144,20 +142,34 @@ def forward(self,  # type: ignore
                                                                           sequence_length,
                                                                           self.num_classes])
         output_dict = {"logits": logits, "class_probabilities": class_probabilities}
+        # We need to retain the mask in the output dictionary
+        # so that we can crop the sequences to remove padding
+        # when we do viterbi inference in self.decode.
+        output_dict["mask"] = mask
+
         if tags is not None:
             loss = sequence_cross_entropy_with_logits(logits,
                                                       tags,
                                                       mask,
                                                       label_smoothing=self._label_smoothing)
             if not self.ignore_span_metric:
-                self.span_metric(class_probabilities, tags, mask)
+                batch_verb_indices = [example_metadata["verb_index"] for example_metadata in metadata]
+                batch_sentences = [example_metadata["words"] for example_metadata in metadata]
+                # TODO (nfliu): This is kind of a hack, consider splitting out part
+                # of decode() to a separate function.
+                batch_bio_predicted_tags = self.decode(output_dict).pop("tags")
```

Comment:
I need a sanity check here...is this cheating? Using `decode()` and passing in the tags is actually "better" than just passing in the `class_probabilities` (since you get the best global tag sequence). The F1 score that I'm getting after running `allennlp evaluate` seems higher than it should be...`89.46 F1`. I figure that using `decode()` should help, but I don't expect a jump of like 4 F1...

Reply:
oops, I'm not ignoring the `V` tag, so the score is incredibly inflated. fix incoming, i verified that the results i get from `allennlp evaluate` matches manually using `allennlp predict`->converting to conll->running the perl file.